plugin/src/test/groovy/org/openbakery/cpd/CpdTaskTest.groovy

```diff
@@ -0,0 +1,116 @@
+package org.openbakery.cpd
+
+import groovy.mock.interceptor.MockFor
+import org.apache.commons.io.FileUtils
+import org.gmock.GMock
+import org.gmock.GMockController
+import org.gradle.api.AntBuilder
+import org.gradle.api.Project
+import org.gradle.testfixtures.ProjectBuilder
+import org.openbakery.CommandRunner
+import org.testng.annotations.AfterMethod
+import org.testng.annotations.BeforeMethod
+import org.testng.annotations.Test
+
+/**
+ * Created by rahul on 7/22/15.
+ */
+class CpdTaskTest {
+    Project project
+    CpdTask cpdTask
+
+    GMockController mockControl
+    CommandRunner commandRunnerMock
+    AntBuilder antMock
+
+    @BeforeMethod
+    void setup() {
+        mockControl = new GMockController()
+        commandRunnerMock = mockControl.mock(CommandRunner)
+        antMock = mockControl.mock(AntBuilder)
+
+        File projectDir =  new File(System.getProperty("java.io.tmpdir"), "gradle-xcodebuild")
+
+        project = ProjectBuilder.builder().withProjectDir(projectDir).build()
+        project.buildDir = new File('build').absoluteFile
+        project.apply plugin:org.openbakery.XcodePlugin
+
+        cpdTask = project.getTasks().getByPath('cpd') as CpdTask
+
+        cpdTask.setProperty("commandRunner", commandRunnerMock)
+    }
+
+    @AfterMethod
+    void cleanUp() {
+        FileUtils.deleteDirectory(project.projectDir)
+    }
+
+/*    @Test
+    void testDownloadAndRun() {
+
+        def destDir = project.gradle.gradleUserHomeDir.absolutePath + "/ios"
+        cpdTask.metaClass.getAnt = { antMock }
+
+        antMock.get([
+                src: 'http://tcpdiag.dl.sourceforge.net/project/pmd/pmd/4.2.5/pmd-bin-4.2.5.zip',
+                dest: new File("${destDir}/pmd-bin-4.2.5.zip")
+        ]).times(1).returns()
+        antMock.get([
+                src: 'https://raw.githubusercontent.com/jkennedy1980/Objective-C-CPD-Language/master/releases/ObjCLanguage-0.0.7-SNAPSHOT.jar',
+                dest: new File("${destDir}/ObjCLanguage-0.0.7-SNAPSHOT.jar")
+        ]).times(1)
+        antMock.unzip(
+                src: "${destDir}/pmd-bin-4.2.5.zip", dest: "${destDir}/tools", overwrite: 'true'
+        ).times(1)
```

Comment:
Take a look here this should help:
https://github.com/openbakery/gradle-xcodePlugin/blob/develop/plugin/src/test/groovy/org/openbakery/oclint/OCLintTaskTest.groovy